vllm/model_executor/layers/quantization/gptq_marlin.py

```diff
@@ -118,19 +180,45 @@ def override_quantization_method(cls, hf_quant_cfg,
                         " faster inference")
         return None
 
+    def get_dynamic_override(
+        self,
+        layer_name: str,
+        key: Optional[str] = None,
+        default_value: Union[int, bool, None] = None
+    ) -> Union[Dict, int, bool, None]:
+        for pattern, pattern_dict in self.dynamic.items():
+            # Negative match: matched modules are excluded from quantized init
+            if pattern.startswith("-:"):
+                if re.match(pattern.removeprefix("-:"), layer_name):
+                    return False
+            # Positive match: matched modules have quant properties overrides
+            # base quant config
+            elif re.match(pattern.removeprefix("+:"), layer_name):
+                if key is None:
+                    return pattern_dict
+                else:
+                    return pattern_dict.get(key, default_value)
+        return default_value
+
     def get_quant_method(
         self, layer: torch.nn.Module, prefix: str
-    ) -> Optional[Union["GPTQMarlinLinearMethod", "GPTQMarlinMoEMethod"]]:
-        if isinstance(layer, LinearBase) or (isinstance(layer, ParallelLMHead)
-                                             and self.lm_head_quantized):
-            return GPTQMarlinLinearMethod(self)
+    ) -> Optional[Union["GPTQMarlinLinearMethod", "GPTQMarlinMoEMethod",
+                        UnquantizedLinearMethod, UnquantizedEmbeddingMethod]]:
+        parallel_lm_head_quantized = isinstance(
+            layer, ParallelLMHead) and self.lm_head_quantized
+        if isinstance(layer, LinearBase) or parallel_lm_head_quantized:
+            # False = skip module, None = no override, else = Positive match
+            if self.get_dynamic_override(layer_name=prefix) == False: 
```

Comment:
@mgoin  `Lint` is being stupid here. We must check for `False` since `None` and `False` are two distinct states here (check comments). How do we force `lint` to pass here?

Ruff (E712)

vllm/model_executor/layers/quantization/gptq_marlin.py:211:16: E712 Avoid equality comparisons to `False`; use `if not self.get_dynamic_override(layer_name=prefix):` for false checks

Reply:
Ok. I was able to force ruff to ignore thie rule for this line using `noqa:`.